carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/utils/android/AndroidUtils.java

```diff
@@ -275,92 +275,262 @@ public static boolean isChecked(final ExtendedWebElement element) {
     }
 
     /**
-     * Scrolls into view in specified container by text only and return ExtendedWebElement
+     * Scrolls into view in specified container by text only and return boolean
      *
-     * @param container  ExtendedWebElement     - works only with resourceId
-     * @param scrollToElement String has to be id, className, text, contentDesc, etc
+     * @param container  ExtendedWebElement - defaults to id Selector Type
+     * @param scrollToElement String defaults to text Selector Type
      * @return ExtendedWebElement
      * <p>
      * example of usage:
-     * ExtendedWebElement res = AndroidUtils.scroll("Hindi", genresTab);
+     * ExtendedWebElement res = AndroidUtils.scroll("News", newsListContainer);
      **/
-    public static ExtendedWebElement scroll(String scrollToElement, ExtendedWebElement container) {
-    	return scroll(scrollToElement, container, SelectorType.TEXT);
+    public ExtendedWebElement scroll(String scrollToElement, ExtendedWebElement container) {
+        return scroll(scrollToElement, container, SelectorType.ID, SelectorType.TEXT);
     }
-    /**
-     * Scrolls into view in specified container and return ExtendedWebElement
-     *
-     * @param container  ExtendedWebElement     - works only with resourceId
-     * @param scrollToElement String has to be id, className, text, contentDesc, etc
-     * @param selectorType  SelectorType can be TEXT, TEXT_CONTAINS, TEXT_STARTS_WITH, ID, DESCRIPTION, DESCRIPTION_CONTAINS, CLASS_NAME
+
+    /** Scrolls into view in a container specified by it's instance (index)
+     * @param scrollToEle - has to be id, text, contentDesc or className
+     * @param scrollableContainer - ExtendedWebElement type
+     * @param containerSelectorType - has to be id, text, textContains, textStartsWith, Description, DescriptionContains
+     *                             or className
+     * @param eleSelectorType -  has to be id, text, textContains, textStartsWith, Description, DescriptionContains
+     *                             or className
+     * @return ExtendedWebElement
+     * <p>
+     * example of usage:
+     * ExtendedWebElement res = AndroidUtils.scroll("News", newsListContainer, AndroidUtils.SelectorType.CLASS_NAME, 1,
+     *                          AndroidUtils.SelectorType.TEXT);
+     **/
+    public static ExtendedWebElement scroll(String scrollToEle, ExtendedWebElement scrollableContainer, SelectorType containerSelectorType,
+                          int containerInstance, SelectorType eleSelectorType) {
+        int maxSearchSwipes = 55; //this value can be changed if necessary
```

Comment:
Should be added as parameter. Default value can be added as Constant

Reply:
Agree on a constant. Re making it as a param, I think it would be better to leave as is. That way we won't overcomplicate things for the end-user. Default of 55 should be plenty.